model/task_lifecycle.go

```diff
@@ -823,7 +819,7 @@ func UpdateBuildAndVersionStatusForTask(taskId string, updates *StatusChanges) e
 		}
 	}
 
-	if buildComplete {
+	if updates.BuildComplete {
```

Comment:
this can be relocated to the previous `if finishedTasks >= len(buildTasks)` block